keras/engine/training.py

```diff
@@ -1040,6 +1040,10 @@ def fit(self, x, y, batch_size=32, nb_epoch=10, verbose=1, callbacks=[],
 
         # prepare display labels
         out_labels = self.metrics_names
+
+        # rename duplicated metrics name
+        out_labels = list(map(lambda x: x[1] + str(out_labels[:x[0]].count(x[1]) + 1) if out_labels.count(x[1]) > 1 else x[1], enumerate(out_labels)))
```

Comment:
I think this line is very long and very difficult to read. It should be rewritten.